src/js/components/FocusedContainer.js

```diff
@@ -60,41 +68,75 @@ export const FocusedContainer = ({
+        } else {
+          // focus was stolen from the container, put focus back where it was
+          attemptFocus(lastFocusRef.current);
+          lastFocusRef.current = document.activeElement;
+        }
+      }
+    };
+
+    const findFirstTabbable = (element) => {
+      for (let index = 0; index < element.childNodes.length; index += 1) {
+        const child = element.childNodes[index];
+        if (isFocusable(child)) return child;
+        const tabbable = findFirstTabbable(child);
+        console.log('first tabbable', tabbable);
+        if (tabbable) return tabbable;
       }
-      // If no focusable child or descendant was found, return false
       return false;
     };
 
-    const focusLastDescendant = (element) => {
-      for (let i = element.childNodes.length - 1; i >= 0; i -= 1) {
-        const child = element.childNodes[i];
-        if (attemptFocus(child) || focusLastDescendant(child)) return true;
+    const findLastTabbable = (element) => {
+      for (let index = element.childNodes.length - 1; index >= 0; index -= 1) {
+        const child = element.childNodes[index];
+        if (isFocusable(child)) return child;
+        const tabbable = findLastTabbable(child);
+        console.log('last tabbable', tabbable);
+        if (tabbable) return tabbable;
       }
       return false;
     };
 
-    const handleTrapFocus = (e) => {
+    // tab natively moves focus, so we need to prevent default to keep the
+    // browser from moving focus.
+    // need to capture at keydown because if we wait for focus event, page could
+    // have already scrolled.
+    const checkTabKey = (e) => {
+      const firstTabbable = findFirstTabbable(container);
+      const lastTabbable = findLastTabbable(container);
+      // Check if the first and last tabbable elements are the same
+      const tabbableElementsEqual = firstTabbable === lastTabbable;
       if (
         !hidden &&
         trapFocus &&
-        ignoreUtilFocusRef.current === false &&
         container &&
         // only perform focus if this is the most recently opened drop
-        roots[roots.length - 1] === container
+        roots[roots.length - 1] === container &&
+        (tabForward(e) || tabBackward(e)) &&
+        (!container.contains(document.activeElement) ||
+          // focus should loop backward to last element
+          (tabBackward(e) && document.activeElement === firstTabbable) ||
+          // focus should loop forward to first element
+          (tabForward(e) && document.activeElement === lastTabbable) ||
+          tabbableElementsEqual)
       ) {
-        if (container.contains(e.target)) {
-          lastFocusRef.current = e.target;
-        } else {
-          focusFirstDescendant(container);
-          if (lastFocusRef.current === document.activeElement) {
-            focusLastDescendant(container);
-          }
-          lastFocusRef.current = document.activeElement;
+        e.preventDefault();
```

Comment:
I think this preventDefault might be preventing more than we want it to. When I tab to the TextInput within the first Drop in the Drop/Multiple story the suggestions drop doesn't open until I press another key


https://github.com/user-attachments/assets/a3e660b7-f5fb-4a5d-a413-5516c8131c5e